src/abilities/Golden-Wyrm.js

```diff
@@ -233,78 +233,139 @@ export default (G) => {
+				has a dynamic health cost. */
+				if (this.creature.health < this._maxTransferAmount + 1) {
+					this.message = 'Not enough health';
+					return false;
+				}
+
 				if (
-					!this.atLeastOneTarget(map, {
+					!this.atLeastOneTarget(this._getHexes(), {
 						team: this._targetTeam,
+						optTest: this._confirmTarget,
 					})
 				) {
 					return false;
 				}
+
 				return true;
 			},
 
-			// 	query() :
 			query: function () {
-				let ability = this;
-				let wyrm = this.creature;
+				const ability = this;
+				const wyrm = this.creature;
 
 				G.grid.queryCreature({
 					fnOnConfirm: function () {
 						ability.animation(...arguments);
 					},
+					optTest: this._confirmTarget,
 					team: this._targetTeam,
 					id: wyrm.id,
 					flipped: wyrm.flipped,
-					hexes: G.grid.getHexMap(wyrm.x - 2, wyrm.y - 2, 0, false, matrices.frontnback2hex),
+					hexes: this._getHexes(),
 				});
 			},
 
-			//	activate() :
 			activate: function (target) {
-				let ability = this;
-				ability.end();
+				this.end();
 
-				let damage = new Damage(
-					ability.creature, // Attacker
-					ability.damages, // Damage Type
-					1, // Area
-					[], // Effects
-					G,
+				// The health transferred is the creature's missing life, capped to 50.
+				const transferAmount = Math.min(
+					target.stats.health - target.health,
+					this._maxTransferAmount,
 				);
-				target.takeDamage(damage);
 
-				//remove frogger bonus if its found
-				ability.creature.effects.forEach(function (item) {
-					if (item.name == 'Offense++') {
-						item.deleteEffect();
-					}
-				});
+				target.heal(transferAmount, false, false);
+
+				/* Damage Golden Wyrm using `.heal()` instead of `.takeDamage()` to apply 
+				raw damage that bypasses dodge, shielded, etc and does not trigger further 
+				effects. */
+				this.creature.heal(-transferAmount, false, false);
+
+				// Rather than individual loss/gain health logs, show a single custom log.
+				this.game.log(
+					`%CreatureName${this.creature.id}% transfers ${transferAmount} health to %CreatureName${target.id}%`,
+				);
+
+				if (this.isUpgraded()) {
+					const regrowthBuffEffect = new Effect(
+						this.title,
+						this.creature,
+						this.creature,
+						'',
+						{
+							alterations: {
+								regrowth: 10,
+							},
+						},
+						G,
+					);
+
+					this.creature.addEffect(
+						regrowthBuffEffect,
+						`%CreatureName${this.creature.id}% gains 10 regrowth points`,
+						'',
+						false,
+						true,
+					);
+				}
+			},
+
+			/**
+			 * Determine the area of effect to query and activate the ability. The area
+			 * of effect are the hexes directly adjacent around the 3-sized creature.
+			 *
+			 * @returns Refer to HexGrid.getHexMap()
+			 */
+			_getHexes() {
+				const map = [
+					[1, 1, 1, 1, 0],
+					[1, 0, 0, 0, 1],
+					[1, 1, 1, 1, 0],
+				];
+				const xOffset = this.creature.y % 2 === 0 ? 0 : 1;
```

Comment:
Can't help but feel there is a helper function that selected hexes while abstracting away the need to check odd/even rows, but was unable to find/understand it.

The hexes are one of the most confusing parts of the codebase to me, still working it out.